### PR TITLE
Template calls

### PR DESCRIPTION
**Pull request by casid**
This is a generalization so that every template can be called from any template.

Tags and Layouts are templates like any other and should be refactored to a more generic behavior.

This feature allows to structure code freely.

For example, page template located at `module1/page1.jte` could extract templates that are only used by this page. Those templates could be now located right next to that page, like this:

`module1/page1-component1.jte`
`module1/page1-component2.jte`

The calls from `module1/page1.jte` would then look like this:

```html
@template.module1.page1-component1()
@template.module1.page1-component2()
```

This is a breaking change and requires a bump to version 2.0.0.

To mitigate the inconvenience for users, the template compiler should be adjusted:
- Fail hard when `@tag.` or `@layout.` is detected when compiling a template
- Generate an error message containing Java code, that the user can use to automatically migrate the currently compiled project

### CODE REVIEW

File: jte/src/test/java/gg/jte/TemplateEngineTest.java [hunk 1]
```diff
@@ -543,6 +543,132 @@ void tagWithVarArgs4() {
...
+    @Test
+    void escapeAt_templateCall() {
+        givenTemplate("mail@@example.com(write me)"); // Escaped with double @
+        thenOutputIs("mail@example.com(write me)");
+    }
+
+    @Test
+    void escapeAt_3x() {
+        givenTemplate("@@@");
+        thenOutputIs("@@");
```
Comment (kelunik) [inline jte/src/test/java/gg/jte/TemplateEngineTest.java:603]: This should throw a syntax error IMO, so all `@` characters are reserved and have to be escaped or might get used in keywords in later releases. That way, there won't be BC breaks if new keywords starting with `@` are introduced.

File: jte/src/test/java/gg/jte/TemplateEngineTest.java [hunk 2]
```diff
@@ -543,6 +543,132 @@ void tagWithVarArgs4() {
...
+        givenTemplate("@parameter(model.hello, model.x), That was an arbitrary template call!");
+        thenOutputIs("One: Hello, two: 42, That was an arbitrary template call!");
+    }
+
+    @Test
+    void templateCall_noConflictWithKeywords_import() {
+        dummyCodeResolver.givenCode("importer.jte", "@param java.lang.String firstParam\n" +
+                "@param int secondParam\n" +
+                "One: ${firstParam}, two: ${secondParam}");
+        givenTemplate("@importer(model.hello, model.x), That was an arbitrary template call!");
```
Comment (kelunik) [inline jte/src/test/java/gg/jte/TemplateEngineTest.java:578]: I think we shouldn't allow arbitrary words right after `@`, as that limits extending the library with new keywords without BC breaks. Maybe something like `@include.importer`?

Comment (casid) [reply jte/src/test/java/gg/jte/TemplateEngineTest.java:578]: Good point. The reason for arbitrary words is mainly to keep `@tag` and `@layout` working without special handling.

Comment (casid) [reply jte/src/test/java/gg/jte/TemplateEngineTest.java]: As we discussed today, I've changed the keyword to `@template`. Thanks for the suggestions, I really like where this is going!

File: jte/src/test/java/gg/jte/TemplateEngineTest.java (hunk 1 above)
Comment (casid) [reply jte/src/test/java/gg/jte/TemplateEngineTest.java]: With the new keyword we don't absolutely need escaping for this PR, so I removed it for now. You're right, and the current implementation is a little bit too rushed to be any good :-D.

Will continue this in https://github.com/casid/jte/issues/125

**Pull request merged**